engine/table/src/main/java/io/deephaven/engine/table/impl/TimeTable.java

```diff
@@ -114,4 +134,309 @@ protected void destroy() {
         super.destroy();
         UpdateGraphProcessor.DEFAULT.removeSource(this);
     }
+
+    private static class SyntheticDateTimeSource implements
+            ColumnSource<DateTime>,
+            DefaultChunkSource.WithPrev<Values>,
+            MutableColumnSourceGetDefaults.LongBacked<DateTime>,
```

Comment:
This `ColumnSource` is immutable, since you are not ever re-using keys in the streaming case.

That means you should use `io.deephaven.engine.table.impl.ImmutableColumnSourceGetDefaults.LongBacked`, delete your `getPrevX` overrides, and your prev fills and gets should delegate to curr (which they already do).